java-checks-test-sources/pom.xml

```diff
@@ -841,12 +841,9 @@
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-compiler-plugin</artifactId>
         <configuration>
-          <release>16</release>
-          <compilerArgs>
```

Comment:
I think we still need this if we want to test preview-features of jdk 17 (pattern in switches)